crates/hyperdrive-math/src/short/max.rs

```diff
@@ -38,6 +38,135 @@ impl State {
             .pow(self.time_stretch())
     }
 
+    /// Use Newton's method with rate reduction to find the amount of bonds
+    /// shorted for a given base deposit amount.
+    pub fn calculate_short_bonds_given_deposit(
+        &self,
+        target_base_amount: FixedPoint<U256>,
+        open_vault_share_price: FixedPoint<U256>,
+        absolute_max_bond_amount: FixedPoint<U256>,
+        maybe_tolerance: Option<FixedPoint<U256>>,
+        maybe_max_iterations: Option<usize>,
+    ) -> Result<FixedPoint<U256>> {
+        let tolerance = maybe_tolerance.unwrap_or(fixed!(1e9));
+        let max_iterations = maybe_max_iterations.unwrap_or(500);
+
+        // The max bond amount might be below the pool's minimum.
+        // If so, no short can be opened.
+        if absolute_max_bond_amount < self.minimum_transaction_amount() {
+            return Err(eyre!("No solvent short is possible."));
+        }
+
+        // Figure out the base deposit for the absolute max bond amount.
+        let absolute_max_base_amount =
+            self.calculate_open_short(absolute_max_bond_amount, open_vault_share_price)?;
+        if target_base_amount > absolute_max_base_amount {
+            return Err(eyre!(
+                "Input too large.
+                target_base_amount={:#?}
+                max_base_amount   ={:#?}",
+                target_base_amount,
+                absolute_max_base_amount
+            ));
+        }
+        // If the absolute max is within tolerance, return it.
+        if absolute_max_base_amount - target_base_amount <= tolerance {
+            return Ok(absolute_max_bond_amount);
+        }
+
+        // Compute a conservative estimate of the bonds shorted & base paid.
+        let mut last_good_bond_amount = self.calculate_approximate_short_bonds_given_base_deposit(
+            target_base_amount,
+            open_vault_share_price,
+        )?;
+
+        // Run Newton's Method to adjust the bond amount.
+        let mut loss = FixedPoint::from(U256::MAX);
+        for _ in 0..max_iterations {
+            // Calculate the current deposit.
+            let current_base_amount =
+                self.calculate_open_short(last_good_bond_amount, open_vault_share_price)?;
+
+            // Calculate the current loss.
+            // TODO: Loss should **always** go down, but it does not.
```

Comment:
Should this be an issue?

Reply:
Actually I can delete this TODO. The two adjustments we make in the `Err` conditions mean that we are not truly doing Newton's Method, and thus the loss is no longer guaranteed to decrease every step.